lib/client.ts

```diff
@@ -290,7 +293,7 @@ const requestWithFilters = (
     .then(paramsOrResponse =>
       paramsOrResponse instanceof ServiceClientResponse
         ? paramsOrResponse
-        : request(paramsOrResponse).catch(error => {
+        : request(paramsOrResponse).catch((error: RequestError) => {
```

Comment:
Do we need to have any runtime checks in here, is it safe to assume we always have `RequestError` (perron code never throws by accident)?